release-notes/swan-lake-release-notes/swan-lake-2201.3.0-release-note.md

```diff
@@ -36,11 +36,142 @@ If you have not installed Ballerina, then download the [installers](/downloads/#
+```ballerina
+public type RefreshTokenGrantConfig record {|
+    @display {
+        iconPath: "Field.icon",
+        label: "clientSecret field",
+        kind: "password"
+    }
+    string clientSecret;
+|};
+```
+
+#### Added support for function pointers with defaultable parameters
+
+Function pointers are now allowed with default values for parameters. Any expression can be used as the default value of a function pointer parameter.
+
+```ballerina
+import ballerina/io;
+
+public function main() {
+    int num1 = 100;
+    int num2 = 50;
+    int num3 = 25;
+    function (int a = 0, int b = 0, int c = 0) returns int total = getSum;
+
+    io:println(total()); // Prints `0`.
+    io:println(total(num1, num2)); // Prints `150`.
+    io:println(total(num1, num2, num3)); // Prints `175`.
+}
+
+function getSum(int num1, int num2, int num3) returns int {
+    return num1 + num2 + num3;
+}
+```
+
 ### Improvements
 
-### Bug fixes
+#### More improvements on working with optional fields
+
+If there is an optional field `T x?;` in a record, the absence of `x` is represented by nil where `T` does not allow nil.
+
+```ballerina
+type Topt record {
+    int x?;
+    int y?;
+};
+
+public function main() {
+    Topt t = {x: 2, y: 4};
+    t.x = ();
+
+    t = {x: 2};
+    var {x: _, y} = t;
+    io:println(y is ()); // true
+
+    int? xOrNil = ();
+    t = {x: xOrNil, y: 5};
+    io:println(t.x is ()); // true
+}
+
+```
+
+#### Made the terminating semicolon optional for module-level declarations
+Previously, a closing semicolon was not allowed after the module-level declarations below.
+- block function body
+- service declaration
+- module class definition
+- module enumeration declaration 
+
+Now, you can optionally end these declarations with a semicolon.
+
+
+### Backward-incompatible changes
+
+#### Disallow using the `-9223372036854775808` unary expression as an integer
+
+Previously, `-9223372036854775808` was allowed to be assigned to an integer in Ballerina. It is now disallowed to comply with the language specification, thus, attempting to assign `-9223372036854775808` to an `int` will now give an error.
+
+```
+int result = -9223372036854775808; // error: '9223372036854775808' is out of range for 'int'
+```
+
+## Compiler API updates
+
+### Improvements
+
+#### Syntax API
+Added a few methods to the `NodeParser` API
+The following methods have been introduced for the `NodeParser` class.
+- `Node parseObjectMember(String text)`
+- `ModulePartNode parseModulePart(String text)`
+- `IntermediateClauseNode parseIntermediateClause(String text, boolean allowActions)`
+
+### Backward-incompatible changes
+
+#### Update NodeFactory methods to allow optional terminating semicolon for module-level declarations
+The methods below in the `NodeFactory` have been updated with an extra parameter for the optional semicolon token.
+- `createServiceDeclarationNode`
+- `createFunctionBodyBlockNode`
+- `createEnumDeclarationNode`
+- `createClassDefinitionNode`
```

Comment:
This also should go under Syntax API right?

Reply:
fixed in [4098632](https://github.com/ballerina-platform/ballerina-release/pull/2180/commits/4098632007e5a40eb3339639fb327422a7889130)